parser/parser.go

```diff
@@ -69,31 +72,25 @@ func ParseFile(r io.Reader) (*File, error) {
 	var b bytes.Buffer
 	var role string
 
-	var lineCount int
-	var linePos int
-
-	var utf16 bool
-
 	var f File
 
 	br := bufio.NewReader(r)
-	for {
-		r, _, err := br.ReadRune()
-		if errors.Is(err, io.EOF) {
-			break
-		} else if err != nil {
-			return nil, err
-		}
 
-		// the utf16 byte order mark will be read as "unreadable" by ReadRune()
-		if isUnreadable(r) && lineCount == 0 && linePos == 0 {
-			utf16 = true
-			continue
-		}
+	var sc scannerDecoder = utf8ScannerDecoder{}
+	if bts, err := br.Peek(2); err != nil {
+		slog.Warn("error parsing byte-order mark", "error", err)
```

Comment:
not returning error here so it falls through and _probably_ fail with `errMissingFrom`